fix: RND-98: correct response schema for tasks.get()

niklub opened the pull request:
Calling `tasks.get()` returns error due to incorrect Task schema

Commit:
fix: RND-98: correct response schema for tasks.get()

Commit:
Merge branch 'develop' into fb-RND-98

Commit:
Modify task serializer with updated_by, file_upload of the correct output type

Commit:
Adjust task api schema with DMTaskSerializer

Commit:
Reformat code

Commit:
Fix AnnotatorsDMField

Commit:
Fix serializer fields in DM

makseq commented on label_studio/data_manager/serializers.py:
```diff
@@ -202,11 +210,51 @@ def update(self, instance, validated_data):
-    annotators = serializers.SerializerMethodField(required=False, read_only=True)
+    predictions = PredictionSerializer(required=False, many=True, default=[], read_only=True)
+    annotations = AnnotationsDMFieldSerializer(required=False, many=True, default=[], read_only=True)
+    drafts = AnnotationDraftDMFieldSerializer(required=False, many=True, default=[], read_only=True)
```
It will break the call of `DataManagerTaskSerializer::get_drafts()` + `LseTaskSerializer::get_drafts()`

https://github.com/HumanSignal/label-studio/assets/501780/a17d6344-f77f-4291-9d6e-e16d1908272f

niklub replied:
Fixed for `predictions` and `drafts` - schemas were rewritten manually on top of `serializers.SerializerMethodField`

makseq commented on label_studio/data_manager/serializers.py:
```diff
@@ -202,11 +210,51 @@ def update(self, instance, validated_data):
-    drafts = serializers.SerializerMethodField(required=False, read_only=True)
-    annotators = serializers.SerializerMethodField(required=False, read_only=True)
+    predictions = PredictionSerializer(required=False, many=True, default=[], read_only=True)
+    annotations = AnnotationsDMFieldSerializer(required=False, many=True, default=[], read_only=True)
```
It _might_ affect the calls of `LseTaskSerializer::get_annotations()` and `LseTaskSerializerForAnnotators::get_annotations()` from LSE

makseq replied:
![image](https://github.com/HumanSignal/label-studio/assets/501780/f5e06d69-e7dd-41c9-8a51-adf686f8a523)

makseq commented on label_studio/data_manager/serializers.py:
```diff
@@ -202,11 +210,51 @@ def update(self, instance, validated_data):
+    predictions = PredictionSerializer(required=False, many=True, default=[], read_only=True)
+    annotations = AnnotationsDMFieldSerializer(required=False, many=True, default=[], read_only=True)
+    drafts = AnnotationDraftDMFieldSerializer(required=False, many=True, default=[], read_only=True)
+    annotators = AnnotatorsDMField(required=False, read_only=True)
```
the same problem as with `get_drafts()` + `LseTaskSerializer::get_annotators()`

makseq commented on label_studio/data_manager/serializers.py:
```diff
@@ -222,7 +270,7 @@ class DataManagerTaskSerializer(TaskSerializer):
     avg_lead_time = serializers.FloatField(required=False)
     draft_exists = serializers.BooleanField(required=False)
-    updated_by = serializers.SerializerMethodField(required=False, read_only=True)
+    updated_by = UpdatedByDMField(required=False, read_only=True)
```
the same problem as with get_drafts()

niklub replied:
Are you sure the problem does appear for this field as well? As it is just as shallow override over `serializers.SerializerMethodField`

makseq commented on label_studio/tasks/openapi_schema.py:
```diff
@@ -34,8 +34,8 @@
     'comment_count': 0,
     'unresolved_comment_count': 0,
     'last_comment_updated_at': '2024-01-15T09:30:00Z',
```
I believe it should be like this: 
`2024-06-18T23:45:46.048490Z`

pls, fix `created_at` and `updated_at` too.

makseq replied:
Or maybe it's better to use the full version of api/tasks response: 
```
{
  "id": 13,
  "predictions": [],
  "annotations": [],
  "drafts": [],
  "annotators": [],
  "inner_id": 2,
  "cancelled_annotations": 0,
  "total_annotations": 0,
  "total_predictions": 0,
  "completed_at": null,
  "annotations_results": "",
  "predictions_results": "",
  "predictions_score": null,
  "file_upload": "6b25fc23-some_3.mp4",
  "storage_filename": null,
  "annotations_ids": "",
  "predictions_model_versions": "",
  "avg_lead_time": null,
  "draft_exists": false,
  "updated_by": [],
  "data": {
    "image": "/data/upload/1/6b25fc23-some_3.mp4"
  },
  "meta": {},
  "created_at": "2024-06-18T23:45:46.048490Z",
  "updated_at": "2024-06-18T23:45:46.048538Z",
  "is_labeled": false,
  "overlap": 1,
  "comment_count": 0,
  "unresolved_comment_count": 0,
  "last_comment_updated_at": null,
  "project": 1,
  "comment_authors": []
}
```

niklub replied:
Fixed

makseq reviewed (changes requested):
Tests are failed in LS and LSE

Commit:
Replace back serializers with SerializerMethodFields

Commit:
Reformat

Commit:
Add datetime examples

Commit:
Add dm task response example

Commit:
Merge branch 'develop' into fb-RND-98

robot-ci-heartex committed:
fix: RND-98: Upgrade with task serializer (merge branch) (#6016)

Hi @niklub!

This PR was
[created](https://github.com/HumanSignal/label-studio/actions/runs/9580652200)
in a response to PRs in upstream repos:
- https://github.com/HumanSignal/label-studio-sdk/pull/252

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: bmartel <brandonmartel@gmail.com>
Co-authored-by: robot-ci-heartex <robot-ci-heartex@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: niklub <nikolai@heartex.com>

niklub commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9611323314)
> Successfully merged:  19 files changed, 242 insertions(+), 181 deletions(-)

niklub committed:
Merge branch 'develop' into 'fb-RND-98'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9611323314

Commit:
Use pyproject from develop
